homeassistant/components/sensor/waze_travel_time.py

```diff
@@ -91,46 +110,93 @@ def icon(self):
     @property
     def device_state_attributes(self):
         """Return the state attributes of the last update."""
-        return {
-            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
-            ATTR_DISTANCE: round(self._state['distance']),
-            ATTR_ROUTE: self._state['route'],
-        }
-
-    def update(self):
-        """Fetch new state data for the sensor."""
-        try:
-            self.waze_data.update()
-            self._state = self.waze_data.data
-        except KeyError:
-            _LOGGER.error("Error retrieving data from server")
-
-
-class WazeRouteData(object):
-    """Get data from Waze."""
-
-    def __init__(self, origin, destination, region):
-        """Initialize the data object."""
-        self._destination = destination
-        self._origin = origin
-        self._region = region
-        self.data = {}
+        if self._state is None:
+            return None
+
+        res = {ATTR_ATTRIBUTION: CONF_ATTRIBUTION}
+        if 'duration' in self._state:
+            res[ATTR_DURATION] = self._state['duration']
+        if 'distance' in self._state:
+            res[ATTR_DISTANCE] = self._state['distance']
+        if 'route' in self._state:
+            res[ATTR_ROUTE] = self._state['route']
+        return res
+
+    def _get_location_from_entity(self, entity_id):
+        """Get the location from the entity state or attributes."""
+        entity = self._hass.states.get(entity_id)
+
+        if entity is None:
+            _LOGGER.error("Unable to find entity %s", entity_id)
+            return None
+
+        # Check if the entity has location attributes (zone)
+        if location.has_location(entity):
+            return self._get_location_from_attributes(entity)
+
+        # Check if device is in a zone (device_tracker)
+        zone_entity = self._hass.states.get("zone.%s" % entity.state)
+        if location.has_location(zone_entity):
+            _LOGGER.debug(
+                "%s is in %s, getting zone location",
+                entity_id, zone_entity.entity_id
+            )
+            return self._get_location_from_attributes(zone_entity)
+
+        # If zone was not found in state then use the state as the location
+        if entity_id.startswith("sensor."):
+            return entity.state
+
+        # When everything fails just return nothing
+        return None
+
+    @staticmethod
+    def _get_location_from_attributes(entity):
+        """Get the lat/long string from an entities attributes."""
+        attr = entity.attributes
+        return "%s,%s" % (attr.get(ATTR_LATITUDE), attr.get(ATTR_LONGITUDE))
```

Comment:
Use string formatting.

Reply:
When I made the update to this file, I don't think I touched this line.  I could have though because I normally deal with python 2.5.  Want me to make these updates?